[Feature]Hudi connector

adzfolc opened the pull request:
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes https://github.com/StarRocks/StarRocksTest/issues/739

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

adzfolc committed:
1. fix hudi table validate
2. fix hudi table column convert
3. fix func name for code rebase
4. add more cases to cover hudi table column convert & validate

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 12 (41.67%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|3|10|30.00%|[276, 277, 278, 279, 281, 282, 284]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 62 / 142 (43.66%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|30|76|39.47%|[78, 79, 118, 131, 149, 152, 155, 157, 190, 192, 193, 195, 197, 198, 199, 200, 201, 206, 211, 269, 270, 311, 333, 335, 350, 351, 365, 366, 369, 370, 371, 372, 373, 374, 375, 376, 378, 379, 380, 381, 383, 384, 389, 390, 458, 459]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|28|60|46.67%|[245, 247, 254, 257, 263, 264, 266, 267, 278, 279, 280, 281, 283, 284, 286, 300, 305, 339, 340, 341, 342, 343, 344, 345, 348, 349, 350, 352, 353, 354, 355, 357]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[265]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|3|66.67%|[206]|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smells

adzfolc commented:
run starrocks_fe_unittest

adzfolc committed:
1. add more comments to code

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 62 / 141 (43.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|30|76|39.47%|[78, 79, 118, 131, 149, 152, 155, 157, 190, 192, 193, 195, 197, 198, 199, 200, 201, 206, 211, 269, 270, 311, 333, 335, 350, 351, 366, 367, 370, 371, 373, 374, 375, 376, 377, 378, 380, 381, 382, 383, 386, 387, 394, 395, 465, 466]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|28|60|46.67%|[246, 248, 255, 258, 264, 265, 267, 268, 279, 280, 281, 282, 284, 285, 287, 301, 306, 340, 341, 342, 343, 344, 345, 346, 349, 350, 351, 353, 354, 355, 356, 358]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[265]|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|

adzfolc committed:
1. feat: hudi connector

adzfolc commented on fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:
```diff
@@ -165,19 +187,28 @@ public Map<String, HiveColumnStats> getTableLevelColumnStats(List<String> column
+                hiveRepository.refreshTableCache(hmsTableInfo);
+            } else {
+                hiveRepository.refreshConnectorTable(resourceName, dbName, tableName);
+            }
```
As HiveTable, internal catalog needs to refreshTableCache, and connector table needs to refreshConnectorTable.

adzfolc commented on fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:
```diff
@@ -285,77 +330,78 @@ private void validate(Map<String, String> properties) throws DdlException {
-                        "to starrocks type [" + column.getPrimitiveType() + "], column name: " + column.getName());
+                        "to starrocks type [" + column.getPrimitiveType() + "], column name: " + column.getName()
+                        + ", starrocks type should be "
+                        + HiveMetaStoreTableUtils.convertHudiTableColumnType(hudiColumn.schema()).toSql());
```
If a column is converted failed, now the exception will show the expcted type.

adzfolc commented on fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:
```diff
@@ -285,77 +330,78 @@ private void validate(Map<String, String> properties) throws DdlException {
-        LogicalType logicalType = avroSchema.getLogicalType();
+        boolean valid = true;
+        Type convertType = HiveMetaStoreTableUtils.convertHudiTableColumnType(avroSchema);
+        // if convertType is ArrayType, so srType must be srType too
```
It is obvious that, if an avroSchema is ArrayType, so the srType must be srType too.So, we can calculate the level of arrays in srType.
eg: level num of Array<Array<Array<int>>> is 3, and the primitive type of srType is int.

adzfolc commented on fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:
```diff
@@ -285,77 +330,78 @@ private void validate(Map<String, String> properties) throws DdlException {
+            // now, the complex types have been reduced to the simple primitive type
+            // we can compare two types by two types' toSql() here
+            valid = srType.toSql().equals(convertType.toSql());
         }
```
As said in comments, varchar type got from avro meta has no length info, so if convertType or srType is varchar, we just need to varify their types, and no need to vertify their length.
And, till now, we can acquire the primitive type of convertType and srType.I come up with a idea that, we can use func `toSql()` to judge whether they are same.

adzfolc commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -227,23 +231,100 @@ public static Type convertColumnType(String hiveType) throws DdlException {
+                } else {
+                    primitiveType = PrimitiveType.VARCHAR;
+                    break;
+                }
```
If logicalType is null, so we transfer to varchar here.

adzfolc commented on fe/fe-core/src/test/java/com/starrocks/external/hive/UtilsTest.java:
```diff
@@ -189,33 +191,64 @@ public void testArrayString() throws DdlException {
+        arraySchema = Schema.createArray(unionSchema);
+        Assert.assertEquals(HiveMetaStoreTableUtils.convertHudiTableColumnType(arraySchema),
+                new ArrayType(ScalarType.createType(PrimitiveType.VARCHAR)));
+    }
```
The cases here, cover the combination of array and primitive types.

adzfolc commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -227,23 +231,99 @@ public static Type convertColumnType(String hiveType) throws DdlException {
         }
     }
 
+    public static Type convertHudiTableColumnType(Schema avroSchema) throws DdlException {
```
> add comments

added

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 119 / 193 (61.66%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[326, 434, 435, 436, 437, 438, 439]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|37|76|48.68%|[118, 131, 190, 192, 193, 195, 197, 198, 199, 200, 201, 206, 211, 269, 270, 311, 333, 335, 350, 351, 366, 367, 370, 371, 373, 374, 375, 376, 377, 378, 380, 381, 382, 383, 386, 387, 395, 465, 466]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[265]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|42|60|70.00%|[246, 248, 255, 258, 264, 265, 267, 268, 279, 280, 281, 282, 284, 285, 287, 301, 306, 346]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. add more comments

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 119 / 193 (61.66%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|37|76|48.68%|[118, 131, 193, 195, 196, 198, 200, 201, 202, 203, 204, 210, 215, 273, 274, 315, 340, 342, 358, 359, 378, 379, 382, 383, 388, 389, 390, 391, 392, 393, 395, 396, 397, 398, 401, 402, 410, 480, 481]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[265]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|42|60|70.00%|[246, 248, 255, 258, 264, 265, 267, 268, 279, 280, 281, 282, 284, 285, 287, 301, 306, 346]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. for decimal type, we check convertType whether is decimal, skip precision

adzfolc commented:
PR Regression Link http://39.101.206.100:8090/view/regression/job/starrocks_pr_regression/578/consoleFull

adzfolc committed:
1. refactor hudi table to adapt connector table and external table, we can differ them by resourceName
2. modify UT, because db name is changed out of catalog, db in default catalog now name is like `default_catalog.db_name`
3. fix code import

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 121 / 198 (61.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|39|81|48.15%|[118, 131, 193, 195, 196, 198, 200, 201, 202, 203, 204, 210, 215, 251, 273, 274, 315, 340, 342, 358, 359, 378, 379, 382, 383, 388, 389, 390, 391, 392, 393, 395, 396, 397, 398, 401, 402, 410, 414, 415, 485, 486]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|42|60|70.00%|[236, 238, 245, 248, 254, 255, 257, 258, 269, 270, 271, 272, 274, 275, 277, 291, 296, 336]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smells

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 121 / 198 (61.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|39|81|48.15%|[118, 131, 193, 195, 196, 198, 200, 201, 202, 203, 204, 210, 215, 251, 273, 274, 315, 340, 342, 358, 359, 378, 379, 382, 383, 388, 389, 390, 391, 392, 393, 395, 396, 397, 398, 401, 402, 410, 414, 415, 485, 486]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|42|60|70.00%|[236, 238, 245, 248, 254, 255, 257, 258, 269, 270, 271, 272, 274, 275, 277, 291, 296, 336]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smells

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 121 / 198 (61.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|39|81|48.15%|[118, 131, 193, 195, 196, 198, 200, 201, 202, 203, 204, 210, 215, 251, 273, 274, 315, 340, 342, 358, 359, 378, 379, 382, 383, 388, 389, 390, 391, 392, 393, 395, 396, 397, 398, 401, 402, 410, 414, 415, 485, 486]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|42|60|70.00%|[236, 238, 245, 248, 254, 255, 257, 258, 269, 270, 271, 272, 274, 275, 277, 291, 296, 336]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smells

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 125 / 202 (61.88%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|39|81|48.15%|[118, 131, 193, 195, 196, 198, 200, 201, 202, 203, 204, 210, 215, 251, 273, 274, 315, 340, 342, 358, 359, 378, 379, 382, 383, 388, 389, 390, 391, 392, 393, 395, 396, 397, 398, 401, 402, 410, 414, 415, 485, 486]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|46|64|71.88%|[237, 239, 246, 249, 255, 256, 258, 259, 271, 272, 273, 274, 276, 277, 279, 293, 298, 342]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smells

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 127 / 206 (61.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|39|81|48.15%|[118, 131, 193, 195, 196, 198, 200, 201, 202, 203, 204, 210, 215, 251, 273, 274, 315, 340, 342, 358, 359, 378, 379, 382, 383, 388, 389, 390, 391, 392, 393, 395, 396, 397, 398, 401, 402, 410, 414, 415, 485, 486]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|48|68|70.59%|[239, 241, 248, 251, 257, 258, 260, 261, 273, 274, 275, 276, 278, 279, 281, 294, 295, 300, 305, 349]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smell

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 128 / 207 (61.84%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|39|81|48.15%|[118, 131, 193, 195, 196, 198, 200, 201, 202, 203, 204, 210, 215, 251, 273, 274, 315, 340, 342, 358, 359, 378, 379, 382, 383, 388, 389, 390, 391, 392, 393, 395, 396, 397, 398, 401, 402, 410, 414, 415, 485, 486]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|49|69|71.01%|[240, 242, 249, 252, 258, 259, 261, 262, 274, 275, 276, 277, 279, 280, 282, 295, 296, 301, 306, 350]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc commented:
http://39.101.206.100:8090/view/regression/job/starrocks_pr_regression/654/

DorianZheng commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -216,23 +223,112 @@ public static Type convertColumnType(String hiveType) throws DdlException {
+                primitiveType = PrimitiveType.DOUBLE;
+                break;
+            case STRING:
+                isString = true;
```
plz refer to hive connector

adzfolc replied:
> plz refer to hive connector

```
case STRING:
    return ScalarType.createDefaultString();
```
Modified

DorianZheng commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -216,23 +223,112 @@ public static Type convertColumnType(String hiveType) throws DdlException {
+                Type type = Utils.convertToArrayType(avroSchema);
+                if (type.isArrayType()) {
+                    return type;
+                }
```
what if this condition is false?

adzfolc replied:
> what if this condition is false?

```
case ARRAY:
    Type type = Utils.convertToArrayType(avroSchema);
    if (type.isArrayType()) {
        return type;
    } else {
        isConvertedFailed = false;
        break;
    }
...
default:
    isConvertedFailed = true;
    break;
    }

if (isConvertedFailed) {
    throw new DdlException(String.format(COLUMN_CONVERTED_FAILED_MSG, avroSchema.getType()));
}

return ScalarType.createType(primitiveType);
```
If `type.isArrayType()` is false, referring to `convertHiveTableColumnType`, type validate should fail here.

adzfolc committed:
1. revert string convert

adzfolc committed:
1. fix code smell

adzfolc committed:
1. revert validColumnType
2. UT adapt convert string type

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 115 / 177 (64.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[120, 133, 195, 197, 198, 200, 202, 203, 204, 205, 206, 212, 217, 253, 275, 276, 317, 342, 344, 360, 361, 488, 489]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|44|66|66.67%|[238, 240, 247, 250, 256, 257, 259, 260, 268, 269, 274, 275, 276, 277, 279, 280, 282, 295, 296, 301, 306, 346]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. split hudi convertToArrayType() from hive/Utils.java to HiveMetaStoreTableUtils

adzfolc commented:
run starrocks_be_build

adzfolc committed:
1. testInputFormat cover all HudiTable.HoodieTableType sub types

adzfolc commented:
run starrocks_be_unittest

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 115 / 177 (64.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[120, 133, 195, 197, 198, 200, 202, 203, 204, 205, 206, 212, 217, 253, 275, 276, 317, 342, 344, 360, 361, 488, 489]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|45|67|67.16%|[238, 240, 247, 250, 256, 257, 259, 260, 268, 269, 274, 275, 276, 277, 279, 280, 282, 295, 296, 301, 306, 350]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

DorianZheng reviewed (dismissed):
LGTM

adzfolc committed:
1. modify func from public to private

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 115 / 177 (64.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[120, 133, 195, 197, 198, 200, 202, 203, 204, 205, 206, 212, 217, 253, 275, 276, 317, 342, 344, 360, 361, 488, 489]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|45|67|67.16%|[238, 240, 247, 250, 256, 257, 259, 260, 268, 269, 274, 275, 276, 277, 279, 280, 282, 295, 296, 301, 306, 350]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc commented:
daily test: http://39.101.206.100:8090/view/Daily/job/master_asan/630/

adzfolc commented:
pr regression: http://39.101.206.100:8090/view/regression/job/starrocks_pr_regression/661/

miomiocat commented:
LGTM

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 115 / 177 (64.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 353, 355, 371, 372, 499, 500]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|45|67|67.16%|[240, 242, 249, 252, 258, 259, 261, 262, 270, 271, 276, 277, 278, 279, 281, 282, 284, 297, 298, 303, 308, 352]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|[42, 43, 44]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

caneGuy commented on fe/fe-core/src/main/java/com/starrocks/connector/hudi/HudiConnector.java:
```diff
@@ -0,0 +1,50 @@
+                metadata = new HudiMetadata(resourceName);
+            } catch (Exception e) {
+                LOG.error("Failed to create hive metadata on [catalog : {}]", catalogName, e);
+                throw e;
```
no exception will be thrown?

adzfolc replied:
> no exception will be thrown?

removed

caneGuy commented on fe/fe-core/src/main/java/com/starrocks/external/hive/Utils.java:
```diff
@@ -23,6 +23,7 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+
```
nit: extra blank line

adzfolc replied:
> nit: extra blank line

removed

adzfolc committed:
1. remove unused exception thrown
2. remove extra blank space

wanpengfei-git commented:
run starrocks_admit_test

adzfolc committed:
1. remove unused imports

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 116 / 175 (66.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 353, 355, 371, 372, 499, 500]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|45|67|67.16%|[240, 242, 249, 252, 258, 259, 261, 262, 270, 271, 276, 277, 278, 279, 281, 282, 284, 297, 298, 303, 308, 352]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smells

wanpengfei-git commented:
run starrocks_admit_test

adzfolc commented:
run starrocks_be_build

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 114 / 173 (65.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 353, 355, 371, 372, 499, 500]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|45|67|67.16%|[240, 242, 249, 252, 258, 259, 261, 262, 270, 271, 276, 277, 278, 279, 281, 282, 284, 297, 298, 303, 308, 352]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 114 / 173 (65.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 353, 355, 371, 372, 499, 500]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|45|67|67.16%|[240, 242, 249, 252, 258, 259, 261, 262, 270, 271, 276, 277, 278, 279, 281, 282, 284, 297, 298, 303, 308, 352]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc commented:
run starrocks_admit_test

adzfolc commented:
run starrocks_admit_test

adzfolc commented:
run starrocks_clang-format

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 114 / 173 (65.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|54|57.41%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 353, 355, 371, 372, 499, 500]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|45|67|67.16%|[240, 242, 249, 252, 258, 259, 261, 262, 270, 271, 276, 277, 278, 279, 281, 282, 284, 297, 298, 303, 308, 352]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. only internal catalog tables need to validate column type

adzfolc committed:
1. convert unsupported types to PrimitiveType.UNKNOWN_TYPE

adzfolc committed:
1. add UT to cover column type convert

adzfolc committed:
1. fix comment typo

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[240, 242, 249, 252, 270, 271, 276, 277, 278, 279, 281, 282, 284, 297, 298, 352]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc committed:
1. fix code smells

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

adzfolc commented:
@Mergifyio backport branch-2.4

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 177 (68.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|10|30.00%|[327, 435, 436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[272]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|33|58|56.90%|[121, 134, 196, 198, 199, 201, 203, 204, 205, 206, 207, 213, 218, 254, 276, 277, 328, 355, 357, 373, 374, 434, 439, 501, 502]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|[45, 46, 47, 62, 63, 64]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|51|67|76.12%|[239, 241, 248, 251, 269, 270, 275, 276, 277, 278, 280, 281, 283, 296, 297, 351]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|[]|
